### ISSUE TITLE
[YAML]: integer keys cannot be un-quoted

### ISSUE THREAD

**Issue by quilicicf** [FasterXML/jackson-dataformats-text#168]
Hi, 

I haven't found a way to serialize a YAML object in this format:

```yaml
200: Whatever
```

It always gets serialized as: 

```yaml
"200": Whatever
```

Online validation tools tell me both versions are perfectly valid.

Upgrading from Jackson 2.9.9 to 2.10.1 led to behavior change and I have to generate the first format so as not to generate a noisy diff for customers of the product I'm working on. 

The enum `YAMLParser#Feature` is empty and no feature from `YAMLGenerator#Feature` seem to change this behavior so I failed when trying to configure my mapper to do it.

Did I miss something?

**Comment by quilicicf** [comment]
You can reproduce with the following code:

```java
private static class Dummy {
        @JsonProperty("200")
        String _200;
        @JsonProperty("400")
        String _400;
    }

    public static void main(final String[] args) throws JsonProcessingException {
        final ObjectMapper mapper = new ObjectMapper(new YAMLFactory().enable(MINIMIZE_QUOTES));
        final Dummy dummy = mapper.readValue("\"200\": 20\n400: 40", Dummy.class);
        final String asString = mapper.writeValueAsString(dummy);
        System.out.println(asString);
    }
```

**Comment by cowtowncoder** [comment]
Right, there is no way to do that and I am not sure there should be way either.

Problem is this: property names are `String` keys. If they were written without quoting, they will be interpreted as numbers. This would make them appear as numbers when reading, changing type essentially.

Situation is, however, different when serializing values like `Map<Integer, WhateverValueType>`: if keys are numbers, they should not be quoted. I think this now works with YAML (as of 2.10); this is one area where there have been changes to try to make it work (problem being that in JSON all keys are Strings; but many other formats allow non-String keys).

Having said all of that, I would not be against yet another `YAMLGenerator.Feature` that would essentially prevent all quoting of property names, if that makes sense. Could be added in 2.11.

**Comment by quilicicf** [comment]
We had the issue in a test that was comparing the output of a model transformation to a locally stored file. 

We managed to fix the comparison by de-serializing -> TreeNode -> re-serializing both values to have a similar formatting for both. Not beautiful but does the work. 

As for the issue itself, I guess a `YAMLGenerator.Feature` would indeed be nice and give more control over the generated format, whatever the dirty implementation details of the underlying POJO. 

I'd love that :)

**Comment by tim-palmer** [comment]
Is this still something you are considering for 2.11?

I have to write a YAML data format with keys which are UUIDs serialized as strings. I'm getting noisy diffs after upgrading to 2.10.1. Previously these were always unquoted, but post-upgrade the UUIDs which start with a digit are quoted, while others are unquoted as before.

Overall the keys cannot be parsed as a number. In theory the generator could work that out but I guess determining that reliably and efficiently might be tricky. So assuming that isn't possible, I'd love to see the proposed feature to prevent quoting of property names.

**Comment by cowtowncoder** [comment]
I do not have plans to work on this; not enough time since I do not actually work on Jackson.
I can help merging contributions, however, if someone wants to try their hand at implementing something.

**Comment by cowtowncoder** [comment]
I don't think this is something I'd want to change/add, closing.
